### ISSUE TITLE
Atom style sheets should use tag selectors only

### ISSUE THREAD

**Issue by TheDataDesigner** [AndcultureCode/AndcultureCode.JavaScript.React.Components#85]
In previous component libraries or projects I've worked on, I've found it helpful to set a rule that the lowest level components (in our case, Atoms) should only consist of a single HTML tag and that the styles for that component use the tag selector instead of a class. For example, the `label` component would be composed of a single `<label>` tag and it's css selection is handled like `label { ... }`. This helps defer complexity to higher layers in the structure, and the lowered specificity can be helpful when styling children in larger components.

Currently, we have a mix of this practice and the `c-component` class pattern. For example, the `anchor` Atom is selected with `a { ... }` while `button` applies a `.c-button { ... }` selection.

The one possible exception to this rule is icons, because there is no dedicated HTML tag for them. Perhaps there are other cases where Atom class names would need to be used to disambiguate two or more distinctly unique component implementations of the same base HTML tag.

I'd suggest that if we implement this change, it happens after #84, since that update would see a number of components currently classified as Atoms move to the Molecule layer where they would retain their class name.

**Comment by wintondeshong** [comment]
@TheDataDesigner all for consistency and happy to see us align our use of selectors across the various "layers". Personally, I'm cool with either direction so long as it is consistent. Sure, I like the idea of fewer class names and I see your point of it "deferring" / suggesting that it go higher up the stack. Suppose the question comes for atoms that don't have a direct 1-to-1 with an html element, right? That an exception and the idea/pattern be to suggest to use tags whenever possible within the atom layer?

Honestly, I had the documentation created before the creation of these actual components and clearly that only ever goes so far (per the need for some of these suggestions). Especially in styling. While it has no bearing on the decision to go tag vs class in approach, without [static analysis](https://github.com/AndcultureCode/AndcultureCode.JavaScript.React.Components/issues/88) to hold folks hands, the rules get bent and broken. I'm going to spin up a separate issue for that so no matter our decision, hopefully we can better uphold it moving forward.

**Comment by TheDataDesigner** [comment]
I definitely agree that having ways to programmatically enforce standards like that would be pretty great. Looking forward to that!

I would also be fine with doing it either way if the rest of the team feels strongly about one way or the other. I _prefer_ classless Atoms because it simplifies in most cases, but it's not a make-or-break kinda deal.

You are right about those exceptions. Atoms without a 1-1 relationship with a specific element would need some kind of name, like the icon component I mentioned. In the past, I've set it up so that most Atomic-level components select the HTML tag itself, and a few exceptions like icons had class names. For me in those cases, having classless components outweighed having absolute consistency since there were very few components that needed that extra class.

The direction we choose to go depends on what we value more: the benefits of having (mostly) classless Atoms, or the consistency of having all Atoms classed.

**Comment by TheDataDesigner** [comment]
Updated the name of the issue to more accurately reflect the discussion.

**Comment by brandongregoryscott** [comment]
I'm fine with going either direction - so long as the pattern is documented. I'm not too familiar on tooling to enforce rules like this, but using a mix of tag selectors and class names might be harder to enforce vs. solely using class names. Just a thought.

**Comment by TheDataDesigner** [comment]
@brandongregoryscott yeah you are right. I think the simpler we can make the rules, the easier they will be to follow.

Another thing I ran into while working on a different issue (#84) is that since we are merging icon variants into their base component, and we're excluding icons from counting towards whether a component should be an Atom or Molecule, some of those Atoms needed wrappers in order to position the icons. The only way to target those wrappers are with classes, so I think it makes sense to just make the rule across the board that we should use class selectors.

I liked the strong tie between Atoms and single HTML tag components, but in this case, maybe it doesn't make sense.
